### DISCUSSION TITLE
Rails 6.1 ships with strict loading to kill n+1 queries, it that possible with Doctrine ORM?

### DISCUSSION THREAD

**Discussion by ruudk** [doctrine/doctrine2#10931] [General]
I saw 2 interesting blog posts today: 

- [Kill N+1 Queries For Good with Strict Loading](https://mattsears.com/articles/2021/05/23/kill-n-plus-one-queries-for-good-with-strict-loading/)
- [Rails 6.1 adds strict_loading to warn lazy loading associations](https://www.bigbinary.com/blog/rails-6-1-adds-strict_loading-to-warn-lazy-loading-associations)

Is it possible to configure Doctrine ORM in a similar way? 

I want to be explicit in my queries. For example, only things that I query in the controller, should be able to be used inside the view. 

Also the new `n_plus_one_only` mode that ships with Rails 7 seems interesting. Detecting these type of n+1 issues, log them, and ideally, prevent them completely.

**Comment by derrabus** [comment]
I'm not aware of a way to achieve that in Doctrine ORM.

**Comment by norkunas** [comment]
Are you looking for #8391 ?

**Comment by dkarlovi** [comment]
> only things that I query in the controller, should be able to be used inside the view.

You're probably looking for view DTO? The idea is you don't pass entities into the view because they (as you've said) allow for N+1. What you can do is pass entity proxies which don't automatically generate additional queries but either require the data to be present or can throw.

We did this in a project in a very similar way, the view objects were entity proxies and all the data required needed to be prepared upfront, the view couldn't create DB requests.

**Comment by ruudk** [comment]
Created a POC to get the Ruby mode working in ORM:
- https://github.com/doctrine/orm/pull/12590
